src/components/Drawer.js

```diff
@@ -167,10 +170,20 @@ class Drawer extends React.Component {
    */
   close = () => {
     this.props.beforeClose();
-    return this._animateComponent({ left: '100%' })
-      .then(() => {
-        this.props.onClose();
-      });
+
+    if (this.props.animateOnClose) {
+      return this._animateComponent({ left: '100%' })
+        .then(() => {
+          this.props.onClose();
+        });
+    } else {
+      // To keep close's api normalized we return a promise just
+      // as the _animateComponent function does above.
```

Comment:
Curious. Who consumes the `close` method's api? Are consumers supposed to call this method directly?

Reply:
I can't say if anyone outside MX does but we do internally some times access the close method directly with a ref so I wanted to make sure it behaved the same for us in those cases.

Reply:
actually that is false statement, we don't use a ref anymore but we do pass close as an argument to children of the drawer if they are a function(function as child)

Reply:
example...

```javascript
<Drawer>
  {({ close }) => {
    <button onClick(close) />
  }}
</Drawer>
```